spec/features/pipelines/cherrypick/cherrypick_pipeline_spec.rb

```diff
@@ -83,122 +86,143 @@
 
-    expect(generated_lines.length).to eq(expected_file_lines.length)
+      expect(fetch_table('table#source_beds')).to eq(table)
 
-    expected_file_lines.each_with_index do |expected_line, index|
-      expect(expected_line).to eq(generated_lines[index])
+      fill_in('SCRC 1', with: '4880000001780')
+      fill_in('DN1S', with: 'DN1S')
+      fill_in('SCRC 2', with: '4880000002794')
+      fill_in('DN5W', with: 'DN5W')
+      fill_in('SCRC 3', with: '4880000003807')
+      fill_in('DN10I', with: 'DN10I')
+      fill_in('DEST 1', with: '4880000020729')
+      fill_in('DN99999F', with: 'DN99999F')
+
+      select 'Custom Type', from: 'destination_plate_types[DN99999F]'
+
+      step 'Wait for verify' do
+        click_button 'Verify'
+        expect(page).to have_content 'Download TECAN file'
+      end
+    end
+
+    step 'Robot verification step 3' do
+      click_link('Download TECAN file')
+      # Tecan file generation is slow. Can probably be sped up, but for the moment...
+      generated_file = DownloadHelpers.downloaded_file("#{batch.id}_batch_DN99999F.gwl")
+      generated_lines = generated_file.lines
+      # Shift off the comment lines
+      generated_lines.shift(2)
+
+      expected_file = <<~TECAN
+        C;
+        C; This file created by user_abc6 on 2018-06-14 11:17:04 +0100
+        C;
+        A;BUFF;;96-TROUGH;1;;49.1
+        D;DN99999F;;Custom Type;1;;49.1
+        W;
+        A;BUFF;;96-TROUGH;2;;49.2
+        D;DN99999F;;Custom Type;2;;49.2
+        W;
+        A;BUFF;;96-TROUGH;3;;49.1
+        D;DN99999F;;Custom Type;3;;49.1
+        W;
+        A;BUFF;;96-TROUGH;4;;49.2
+        D;DN99999F;;Custom Type;4;;49.2
+        W;
+        A;BUFF;;96-TROUGH;5;;49.1
+        D;DN99999F;;Custom Type;5;;49.1
+        W;
+        A;BUFF;;96-TROUGH;6;;49.2
+        D;DN99999F;;Custom Type;6;;49.2
+        W;
+        C;
+        A;DN1S;;ABgene 0765;1;;15.9
+        D;DN99999F;;Custom Type;1;;15.9
+        W;
+        A;DN1S;;ABgene 0765;2;;15.8
+        D;DN99999F;;Custom Type;2;;15.8
+        W;
+        A;DN10I;;ABgene 0765;1;;15.9
+        D;DN99999F;;Custom Type;3;;15.9
+        W;
+        A;DN10I;;ABgene 0765;2;;15.8
+        D;DN99999F;;Custom Type;4;;15.8
+        W;
+        A;DN5W;;ABgene 0765;1;;15.9
+        D;DN99999F;;Custom Type;5;;15.9
+        W;
+        A;DN5W;;ABgene 0765;2;;15.8
+        D;DN99999F;;Custom Type;6;;15.8
+        W;
+        C;
+        C; SCRC1 = DN1S
+        C; SCRC2 = DN10I
+        C; SCRC3 = DN5W
+        C;
+        C; DEST1 = DN99999F
+      TECAN
+
+      expected_file_lines = expected_file.lines
+      # Shift off the comment lines
+      expected_file_lines.shift(2)
+
+      expect(generated_lines.length).to eq(expected_file_lines.length)
+
+      expected_file_lines.each_with_index do |expected_line, index|
+        expect(generated_lines[index]).to eq(expected_line), "Error on line #{index + 2} in #{expected_file}"
```

Comment:
This assertion was the wrong way round, which caused misleading error messages. I've also added spitting out the line number, and body of the response to make errors easier to diagnose.